core/shared/src/main/scala/spire/algebra/Gcd.scala

```diff
@@ -0,0 +1,25 @@
+package spire.algebra
+
+import annotation.tailrec
+import scala.{specialized => sp}
+
+import spire.syntax.eq._
+import spire.syntax.euclideanRing._
+
+trait Gcd[@sp(Int, Long, Float, Double) A] extends Any {
+  def gcd(a: A, b: A): A
+  def lcm(a: A, b: A): A
+
+  @tailrec final def euclid(a: A, b: A)(implicit eq: Eq[A], er: EuclideanRing[A]): A =
```

Comment:
We can probably move this to the companion object, since it doesn't actually use `gcd` or `lcm`.